src/ray/util/event.cc

```diff
@@ -312,6 +367,33 @@ void RayEvent::SendMessage(const std::string &message) {
     event.mutable_custom_fields()->insert(mp.begin(), mp.end());
 
     EventManager::Instance().Publish(event, custom_fields_);
+  } else if (is_export_event) {
```

Comment:
Hmm why does it have to be separated from the API above? can we just reuse code above?

Reply:
Were you referring to reusing code to generate the id? This logic populates the `ExportEvent` message with different fields than `Event`, so the other fields are formed differently